armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -0,0 +1,492 @@
+# Copyright 2019 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""enable component-wise axial expansion for assemblies and/or a reactor"""
+
+from statistics import mean
+from numpy import array
+from armi import runLog
+from armi.reactor.flags import Flags
+
+
+class AxialExpansionChanger:
+    """
+    Axially expand or contract assemblies or an entire core.
+
+    Useful for fuel performance, thermal expansion, reactivity coefficients, etc.
+    """
```

Comment:
It's a good idea to have an attributes section here that describes the "linked" and "expansionData" public attributes so someone knows what they represent and how to interact with them.

```
Attributes
----------------
linked : bool
   Blah

expansionData : whatever object type
   Blah
```

Reply:
Addressed.

https://github.com/terrapower/armi/pull/606/commits/acea48881626894854c2aacb88b6c56873b8d29a

Reply:
Might be able to just have the class name in the brackets if it's part of the same module, but looks good thanks! I personally wouldn't add the pylint comment because that's going to get rendered in the docs and be weird.

Reply:
@jakehader, addressed.
https://github.com/terrapower/armi/pull/606/commits/8d3de41d260b45aea7b9e1ce17b2ac0b5182f914